app/src/main/java/org/dhis2/data/forms/dataentry/DataEntryAdapter.java

```diff
@@ -261,27 +251,14 @@ public void swap(@NonNull List<FieldViewModel> updates) {
 
         diffResult.dispatchUpdatesTo(this);
 
-        if (nextFocusPosition != -1 && currentFocusPosition == lastFocusPosition && nextFocusPosition < viewModels.size())
+        if (nextFocusPosition != -1 && currentFocusPosition == lastFocusPosition && nextFocusPosition < viewModels.size()) {
             currentFocusUid.setValue(viewModels.get(nextFocusPosition).uid());
-        else if (currentFocusPosition != -1 && currentFocusPosition < viewModels.size())
+        } else if (nextFocusPosition == viewModels.size()) {
+            currentFocusUid.setValue("");
```

Comment:
is it needed?

Reply:
Yes, because when the focus is on the last field and the currentFocusUid is not updated, the last will request focus and continue to show the keyboard to the user. Basically this is done to close the keyboard on the last text field of the form.